### ISSUE TITLE
Constructor with parameters support

### ISSUE THREAD

**Issue by hugodantas** [litedb-org/LiteDB#1132]
Hi David.

Is there any plan to support parametrized constructors like MongoDB?
I made a test with MongoDB and classes with parametrized constructors and it can (de)serialize... Is there any possibility to do the same with LiteDB?
Thank you so much.

================ PT/BR ==============

Olá David.
Por acaso existe algum planejamento para suportar classes com construtores com parâmetros tal como já funciona no MongoDB?
Eu fiz uns testes aqui com o MongoDB usando uma classe com construtor parametrizado e vi que ele conseguiu serializar e deserializar sem customizar o mapeamento.
Edit: Percebi que o problema não está em serializar uma classe com parâmetros, mas em deserializar do banco pelo GetCollection<T>().FindAll(), por exemplo.

Segue abaixo o teste:

https://gist.github.com/hugodantas/993801de20e4d4fcdc19c02eb38c39f7

Obrigado.

**Comment by hugodantas** [comment]
Is there any other solution to handle this issue?
Reflection isn't performatic to solve this problem...

Thank you!

**Comment by HofmeisterAn** [comment]
I agree. Especially for 3rd party libraries, you cannot control the implementation of their constructs.

**Comment by JKamsker** [comment]
This is fixed on the `dev` branch. `dev` ships as the LiteDB 6.x prereleases.

Please try the latest prerelease (currently **6.0.0-prerelease.81**):

```
dotnet add package LiteDB --prerelease
```

If the problem still happens on the prerelease, comment here with the version you used and a repro, and we'll reopen the issue.
